src/mots/module.py

```diff
@@ -0,0 +1,356 @@
+        valid path and a valid name and machine name,  and then run the same validation
+        on all submodules.
+
+        :param errors: a list of errors to append to
+        :rtype: list
+        """
+        errors = errors or []
+
+        if not self.machine_name.strip():
+            errors.append("Module has a blank machine_name.")
+
+        if " " in self.machine_name:
+            errors.append(f"Machine name {self.machine_name} contains white space.")
+
+        if not self.calculate_paths():
+            errors.append(f"No valid paths were found in {self.machine_name}.")
+
+        # # TODO do this validation against BMO/mock client
+        # if self.owners:
+        #     for owner in self.owners:
+        #         logger.debug(f"Parsing owner {owner}...")
+        #         owner = parse_user_string(owner)
+        #         if not owner["email"]:
+        #             errors.append(f"No valid email found for owner {owner['name']}")
+        # if self.peers:
+        #     for peer in self.peers:
+        #         logger.debug(f"Parsing peer {peer}...")
+        #         peer = parse_user_string(peer)
+        #         if not peer["email"]:
+        #             errors.append(f"No valid email found for peer {peer['name']}")
+
+        if self.submodules:
+            for submodule in self.submodules:
+                return submodule.validate(errors=errors)
+
+        return errors
+
+
+def ls(modules: list[Module]):
+    """Print a list of given modules.
+
+    :param modules: a list of :class:`Module` instances
+    """
+    print(modules)
+
+
+def show(modules: list[Module], module: str):
+    """Show details for a particular module.
+
+    :param modules: a list of :class:`Module` instances
+    :param module: the `machine_name` of a given :class:`Module`
+
+    :raises ValueError: when no module matches the given machine name
+    """
+    modules_by_name = {}
+    for _module in modules:
+        modules_by_name[_module["machine_name"]] = _module
+    if module not in modules_by_name:
+        raise ValueError(f"{module} not found.")
+    else:
+        print(modules_by_name[module])
+
+
+def extract_people(module):
+    """Return a list of people that are in a module or submodule."""
+    people_keys = ["owners", "peers"]  # "owners_emeritus", "peers_emeritus"
+    people = []
+    for key in people_keys:
+        if key in module and module[key]:
+            logger.debug(f"Extracting people from {module[key]} ({key})")
+            people += module[key]
+    return people
+
+
+def clean(file_config: FileConfig, write: bool = True):
+    """Clean and re-sort configuration file.
+
+    Load condiguration from disk, sort modules and submodules by `machine_name`. If
+    there is no valid `machine_name`, generate one. Write changes to disk.
+
+    :param file_config: an instance of :class:`FileConfig`
+    :param write: if set to `True`, writes changes to disk.
+    """
+    from mots.directory import Directory
```

Comment:
This is done here to prevent circular import. There might be an alternate way to structure the imports but as these two modules are currently laid out, this import needs to be here.